samples/sensor/tmp116/src/main.c

```diff
@@ -14,13 +14,29 @@ void main(void)
 {
 	const struct device *dev;
 	struct sensor_value temp_value;
+
+	/* offset to be added to the temperature
+	 * only supported by TMP117
+	 */
+	struct sensor_value offset_value;
 	int ret;
 
 	dev = device_get_binding(DT_LABEL(DT_INST(0, ti_tmp116)));
 	__ASSERT(dev != NULL, "Failed to get TMP116 device binding");
 
 	printk("Device %s - %p is ready\n", dev->name, dev);
 
+	/*
+	 * if an offset of 2.5 oC is to be added, set val1 = 2 and val2 = 5
+	 */
+	offset_value.val1 = 0;
+	offset_value.val2 = 0;
+	ret = sensor_attr_set(dev, SENSOR_CHAN_AMBIENT_TEMP,
+			      SENSOR_ATTR_OFFSET, &offset_value);
+	if (ret) {
+		printk("sensor_attr_set failed ret %d\n", ret);
+		return;
```

Comment:
This sample will simply return early when used with `tmp116`, because sensor_attr_set will return `-EINVAL`.

Reply:
I have corrected this.